### ISSUE TITLE
Debugger breakpoint takes a while to stop

### ISSUE THREAD

**Issue by cavalos0086** [Microsoft/vscode-recipes#193]
# Issue:

When I use the launch docker debugger method, the debugger starts but the breakpoints don't work. I have to wait about a min for the breakpoints to work. 

# Information:
Vscode version: `1.31.1 (1.31.1)`
Docker desktop community version `2.0.0.3(31259)

### Launch configuration:
```
{
      "type": "node",
      "request": "launch",
      "name": "Launch in Docker",
      "preLaunchTask": "tsc-watch",
      "protocol": "auto",
      "runtimeExecutable": "npm",
      "runtimeArgs": [ "run", "docker-debug" ],
      "port": 5858,
      "restart": true,
      "timeout": 60000,
      "localRoot": "${workspaceFolder}",
      "remoteRoot": "/server",
      "outFiles": [
        "${workspaceFolder}/dist/**/*.js"
      ],
      "console": "integratedTerminal",
      "internalConsoleOptions": "neverOpen"
    },
```

### tasks configuration:
```
"tasks": [
    {
      "label": "tsc-watch",
      "isBackground": true,
      "type": "typescript",
      "tsconfig": "src/tsconfig.json",
      "option": "watch",
    }
  ]
```

**Comment by auchenberg** [comment]
How does your `npm run docker-debug` script look?

**Comment by cavalos0086** [comment]
`"docker-debug": "docker-compose up",`
and the docker-compose file:
```
version: "2"

services:
  web:
    build: .
    command: npm run debug
    volumes:
      - ./dist:/server/dist
    ports:
      - "3000:3000"
      - "5858:5858"
```

**Comment by cavalos0086** [comment]
I was able to find the answer to this issue. I was using the `timeout` in the launch configuration in the wrong way. 

Vscode takes my launch configuration and runs the `timeExecutables` then it waits for `timeout` seconds to attach to the debugger because this is the time it thinks it will take for the executable to finish. 

My launch configuration only brings up the container up so I was able to reduce the time to `1s` instead and it worked well now. However, another solution would be to create a `preLaunchTaks` and have this tasks bring up the container then I don't risk being timeout when the node env is ready. 

I wil close the issue now. 

Thanks
